### PR TITLE
`Development:` Add sphinx gradle plugin

### PR DESCRIPTION
**Pull request by manuelmanso**
#### General
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Adds a sphinx plugin for gradle that generates the docs without having to install/run anything (python/pip) locally. Simply run `./gradlew sphinx` to generate the docs locally.

### Description
Adds the plugin to build.gradle, points to the source directory (/docs) and adds a comment on how to run it.

This is the plugin: https://trustin.github.io/sphinx-gradle-plugin/

####  Review
- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -411,3 +416,4 @@ pitest {
 // 9) Verify code coverage (after tests):           ./gradlew jacocoTestCoverageVerification
 // 10) Detects uses of legacy code:                 ./gradlew modernizer -x webapp
 // 11) Run PIT Mutation Tests:                      ./gradlew pitest -x webapp
+// 11) Generate docs:                               ./gradlew site
```
Comment (Strohgelaender) [inline build.gradle:419]:
```suggestion
// 12) Generate docs:                               ./gradlew site
```

**Review by Strohgelaender — APPROVED** [review]
Tested locally, works.

**Review by Marvin-Vitcu — DISMISSED** [review]
tested locally, worked as intended

**Review by Strohgelaender — APPROVED** [review]

**Comment by dfuchss** [comment]
I have to ask, because I'm not sure: How stable/mature is the Plugin? It hasn't been updated for a while and afaik some new major release of sphinx has been released some weeks ago.

**Review by Hialus — CHANGES_REQUESTED** [review]
> I have to ask, because I'm not sure: How stable/mature is the Plugin? It hasn't been updated for a while and afaik some new major release of sphinx has been released some weeks ago.

Thats a good remark. I just took a deeper look into it:
- Artemis currently uses Sphinx 4.2.0 (we might want to look into updating this)
- The newest Sphinx version is 5.0.2
- The sphinx-binary project (which is the basis for this plugin) is in version 0.8.2 (released 11.02.2021) and uses Sphinx version 3.4.3!

So even now the plugin is on an outdated major version of Sphinx. If we update to Sphinx 5 in the future the plugin would be 2 major versions behind. Considering that there has not been an update to the plugin in over a year for Sphinx 4/5, it is safe to say that the plugin has been abandoned before it even reached version 1.0.0.

My conclusion is therefore that this PR should not get merged.

**Comment by manuelmanso** [comment]
Thanks for the input and research, I'll close it, indeed there's not much point if it's supporting only outdated version  👍

**Pull request closed without merging**